### PR TITLE
Remove the address_configured flag on tcp::resolver::query

### PR DESCRIPTION
**Pull request by robhancocksed**
The default behavior for tcp::resolver::query uses the
address_configured flag, which only returns addresses if a non-loopback
address is available on the system. If this is called before a real network
interface is brought up, then resolution will fail and the server won't
be functional, even for requests on the loopback interface.

Explicitly set the flags to default so that the address_configured flag
is not specified. This allows the server to start up normally even when
the system has no active network interfaces.

### CODE REVIEW

File: Release/src/http/listener/http_server_asio.cpp [hunk 1]
```diff
@@ -520,7 +520,9 @@ void hostport_listener::start()
     auto& service = crossplat::threadpool::shared_instance().service();
     tcp::resolver resolver(service);
     // #446: boost resolver does not recognize "+" as a host wildchar
-    tcp::resolver::query query = ("+" == m_host) ? tcp::resolver::query(m_port) : tcp::resolver::query(m_host, m_port);
+    tcp::resolver::query query = ("+" == m_host) ?
```
Comment (garethsb-sony) [inline Release/src/http/listener/http_server_asio.cpp]: I was going to make this same PR!

The following would be shorter if it compiled cleanly everywhere?

```diff
-    tcp::resolver::query query = ("+" == m_host) ? tcp::resolver::query(m_port) : tcp::resolver::query(m_host, m_port);
+    tcp::resolver::query query = ("+" == m_host) ? tcp::resolver::query(m_port, {}) : tcp::resolver::query(m_host, m_port, {});
```

**Comment by BillyONeal** [comment]
Is there any way to test this? I don't really know what this code is doing and without tests I'm worried we will regress this scenario for you in the future.

**Comment by garethsb-sony** [comment]
This affects Boost.ASIO-based `http_listener` on a machine which doesn't currently have a non-loopback address. I suspect that makes it hard to test without the user intervention of disabling network interfaces.

Another report, with the same fix, here: https://stackoverflow.com/a/5976632

**Comment by BillyONeal** [comment]
Right, I trust the fix, I don't trust myself or a future maintainer to not unfix :)

**Comment by garethsb-sony** [comment]
How can we help? :-)

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**